### ISSUE TITLE
[kube-prometheus-stack] Missing support for `alertmanagerconfigs.monitoring.coreos.com` v1beta1 CRD

### ISSUE THREAD

**Issue by patrickshan** [prometheus-community/helm-charts#4827]
### Is your feature request related to a problem ?

There is no support for `alertmanagerconfigs.monitoring.coreos.com` `v1beta1` CRD inside kube-prometheus-stack at the moment but only `v1alpha1`.

It seems to be able to solve another issue https://github.com/prometheus-community/helm-charts/issues/2711 when this is fixed

### Describe the solution you'd like.

Add support for v1beta1 version support for `alertmanagerconfigs.monitoring.coreos.com` 

### Describe alternatives you've considered.

NONE

**Comment by siegenthalerroger** [comment]
I'm also running into the same issue, the CRD is also not included in the "prometheus-operator-crds" chart.

**Comment by choeffer** [comment]
Can confirm. The issue is that https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/charts/crds/crds/crd-alertmanagerconfigs.yaml follows the wrong upstream folder.
It should point to https://github.com/prometheus-operator/prometheus-operator/blob/main/example/prometheus-operator-crd-full/monitoring.coreos.com_alertmanagerconfigs.yaml where the v1beta1 is included.
Currently it follows https://github.com/prometheus-operator/prometheus-operator/blob/main/example/prometheus-operator-crd/monitoring.coreos.com_alertmanagerconfigs.yaml 

So the fix is to take the CRDs from https://github.com/prometheus-operator/prometheus-operator/blob/main/example/prometheus-operator-crd-full/ . It might be worth checking if other CRDs are effected by this as well, as both subfolders (prometheus-operator-crd-**full** and prometheus-operator-crd) exist upstream.

**Comment by siegenthalerroger** [comment]
The same issue exists with the separate prometheus-operator-crds chart. I'm slightly confused what the intended use-case for the admission-webhook chart is if there's no managed way of installing the v1beta1 chart. 

Of course doing it manually is easy enough but still.

**Comment by jkroepke** [comment]
Hi,

My last update was that the v1beta1 AlertmanagerConfig CRD does not provide any benefits. I looked into this more closely while integrating it (https://github.com/prometheus-community/helm-charts/pull/4526) and found that the served property is set to false for v1beta1. This means that even if an AlertmanagerConfig in v1beta1 is applied via kubectl, it will be converted to v1alpha1 before being stored in etcd.

In conclusion, using v1beta1 has no practical use case. It appears to be in an alpha state—existing but not intended for actual use.

**Comment by snowdrop4** [comment]
There are functional differences between v1alpha1 and v1beta1.

[v1alpha1](https://doc.crds.dev/github.com/prometheus-operator/kube-prometheus/monitoring.coreos.com/AlertmanagerConfig/v1alpha1@v0.7.0) doesn't have the `timeIntervals` key, unlike [v1beta1](https://docs.okd.io/latest/rest_api/monitoring_apis/alertmanagerconfig-monitoring-coreos-com-v1beta1.html).

**Comment by jkroepke** [comment]
Prometheus Operator Maintainers says that v1beta1 CRD version is not ready to use.

ref: https://github.com/prometheus-operator/prometheus-operator/issues/7087#issuecomment-2721286949

**Comment by snowdrop4** [comment]
Ah, thank you.
